streams/src/main/java/org/apache/kafka/streams/state/internals/CanSendOldValues.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+public interface CanSendOldValues {
```

Comment:
My understanding is that with the wrapping of `metered` -> `caching` -> `inner` stores, we need a way to tell the `inner` store, for example RocksDB to turn on sending old values from `init` functions of `KStreamWindowAggregate`, etc, without exposing the function in public APIs.

In that case, could we modify the wrapping as:

1. `caching` -> `metered` -> `inner` such that `metered` is only metering the underlying store, and we may consider adding other metering sensors to the caching layer separately, see https://issues.apache.org/jira/browse/KAFKA-3183.

2. Then we can rename this class to `CachedStateStore` and only have `CachingKeyValueStore` and `CachingWindowedStore` (see my previous comment for the second class) extend this interface.

3. In `KStreamWindowedAggregate` etc, call

```
((CachedStateStore) store).enableForwardingOldValues();
```

Is it cleaner?